scipy/stats/_continuous_distns.py

```diff
@@ -1846,6 +1847,109 @@ def _entropy(self, c):
 weibull_min = weibull_min_gen(a=0.0, name='weibull_min')
 
 
+class truncweibull_min_gen(rv_continuous):
+    r"""A truncated Weibull minimum continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `truncweibull_min` is:
+
+    .. math::
+
+        f(x, a) = c x^{c-1} \exp(-x^c)
+
+    for :math:`x > a`, :math:`c > 0`.
+
+    `truncweibull_min` takes :math:`a` and :math:`b` as shape parameters.
+
+    Notice that the truncation value, :math:`a`, is defined in standardized
+    form::
+
+        a = (u - loc)/scale
+
+    where :math:`u` is the specific truncation value.
+
+    %(after_notes)s
+
+    %(example)s
+
+    """
+    def _argcheck(self, a, c):
+        self.a = a
+        return (a >= 0.) & (c > 0.)
+
+    def _pdf(self, x, a, c):
+        return c*pow(x, c-1)*np.exp(-pow(x, c) + pow(a, c))
+
+    def _logpdf(self, x, a, c):
+        logpdf = np.log(c) + sc.xlogy(c - 1, x) - pow(x, c) + pow(a, c)
+        return logpdf
+
+    def _cdf(self, x, a, c):
+        return -sc.expm1(-pow(x, c) + pow(a, c))
+
+    def _sf(self, x, a, c):
+        return np.exp(-pow(x, c) + pow(a, c))
+
+    def _logsf(self, x, a, c):
+        return -pow(x, c) + pow(a, c)
+
+    def _ppf(self, q, a, c):
+        return pow(pow(a, c) - sc.log1p(-q), 1.0/c)
+
+    def _a_est(self, x, c):
+        x_r = x.min()
+        f_r = (1. - 0.3)/(len(x) + 0.4)
+
+        if pow(-np.log(1 - f_r), 1./c) > x_r:
+            return 0
+        else:
+            return (np.log(1. - f_r) + x_r**c)**(1./c)
+
+    @contextmanager
+    def _mle_mod(self):
+        """Override penalized_nnlf method in a context manager."""
+        penalized_nnlf = self._penalized_nnlf
+        try:
+            self._penalized_nnlf = self._penalized_nnlf_mod
+            yield
+        finally:
+            self._penalized_nnlf = penalized_nnlf
+
+    def _penalized_nnlf_mod(self, theta, x):
+        """
+        Ignore provided `a` and estimate it using an alternative method. This
+        modifies the resulting ML function sent to the optimizer.
+
+        """
+        loc, scale, (_, c) = self._unpack_loc_scale(theta)
+        theta = (self._a_est((x - loc)/scale, c), c, loc, scale)
+        return super(truncweibull_min_gen, self)._penalized_nnlf(theta, x)
+
+    def fit(self, data, *args, **kwds):
+
+        f0 = (kwds.get('f0', None) or kwds.get('fa', None) or
```

Comment:
Default for `get` is already `None` but you are not using the actual values anyways so you don't need to get the values. Instead you can convert these to explicit bools 

```Python
return any([x in kwds for x in ['f0', 'fa', 'fix_a']])
```
is a definite True or False then below you can just check 

```Python
if f0 or (floc  and fscale):
    ...
```

which I think is much more readable.

Reply:
That is true for "f0". However, just trying to be consistent with the rest of the code base here, e.g. see lognorm_gen.fit().

For floc (and fscale) it is necessary to explicitly check for None, since 0 is a valid input.

Reply:
Yeah looks like the `fit` override is gone.